airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -472,9 +472,9 @@ def latest_item_select_object(cls, dag_id):
         return select(cls).where(cls.dag_id == dag_id).order_by(cls.created_at.desc()).limit(1)
 
     @classmethod
-    @provide_session
+    @provide_readonly_session
     def get_latest_serialized_dags(
-        cls, *, dag_ids: list[str], session: Session = NEW_SESSION
+        cls, *, dag_ids: list[str], session: Session = NEW_READONLY_SESSION
```

Comment:
Nit: I don't think we need `NEW_READONLY_SESSION` to be distinct from `NEW_SESSION` -- as the default doesn't actually do anything

Reply:
(I know this is draft and to go with the AIP, so I haven't looked much beyond this)